btcutil/psbt/go.mod

```diff
@@ -24,3 +24,5 @@ require (
 replace github.com/btcsuite/btcd/btcutil => ../
 
 replace github.com/btcsuite/btcd => ../..
+
+replace github.com/btcsuite/btcd/chaincfg/chainhash => ../../chaincfg/chainhash
```

Comment:
To avoid making the circular dependency problem between some of the packages more annoying, I think we should first push the change in `chaincfg/chainhash`, push a new tag, then update these two `go.mod` files to the latest tag.
I wanted to get rid of the other local replace directives as well in another PR, just never got to it...

Reply:
Should I make a separate PR that only adds the `DoubleHashRaw` function?

I can definitely break this up into multiple PRs.

Reply:
Yes, to avoid needing more local replace directives, this needs to be done in two PRs.

Reply:
Yeah ordering wise we'll need to:
  1. Remove this section. 
  2. Merge the PR. 
  3. Tag a new version. 
  4. Make a new PR that actually _uses_ the new functions.